tests_e2e/tests/lib/shell.py

```diff
@@ -46,9 +46,17 @@ def run_command(command: Any, shell=False) -> str:
     NOTE: The command's stdout and stderr are read as text streams.
     """
     process = Popen(command, stdout=PIPE, stderr=PIPE, shell=shell, text=True)
+    timer = threading.Timer(15 * 60, process.kill)  # Kill process after timeout
 
-    stdout, stderr = process.communicate()
+    try:
+        timer.start()
```

Comment:
start() should probably be outside the try statement (if it throws, I think we do not want to call cancel())